src/Functions/FunctionsStringArray.h

```diff
@@ -271,6 +292,42 @@ class SplitByCharImpl
             throw Exception("Illegal separator for function " + getName() + ". Must be exactly one byte.", ErrorCodes::BAD_ARGUMENTS);
 
         sep = sep_str[0];
+
+        if (arguments.size() > 2)
+        {
+            std::optional<Int64> max_split_opt = std::nullopt;
```

Comment:
It seems if we know that value is constant, and number, then we can get column field using for example column function `get`, after that we need to first try to get field type, if its Int64 and it is positive take it as max_split if it negative throw exception, if field type is not Int64 then it will hold UInt64, just use it as max_split.

Reply:
done